ow_faults/src/FaultInjector.cpp

```diff
@@ -93,24 +106,61 @@ void FaultInjector::antennaePanFaultCb(const std_msgs::Float64& msg){
   publishAntennaeFaults(msg, 
                         m_faults.ant_pan_encoder_failure, 
                         m_faults.ant_pan_effort_failure, 
-                        m_faultPanValue, m_fault_ant_pan_pub );
+                        m_fault_pan_value, m_fault_ant_pan_remapped_pub );
 }
 
 void FaultInjector::antennaeTiltFaultCb(const std_msgs::Float64& msg){
   publishAntennaeFaults(msg, 
                         m_faults.ant_tilt_encoder_failure, 
                         m_faults.ant_tilt_effort_failure, 
-                        m_faultTiltValue, m_fault_ant_tilt_pub );
+                        m_fault_tilt_value, m_fault_ant_tilt_remapped_pub );
+}
+
+float FaultInjector::getRandomFloatFromRange( float min_val, float max_val){
+  return min_val + (max_val - min_val) * (rand() / static_cast<float>(RAND_MAX));
 }
 
-void FaultInjector::setPowerTemperatureFaultValue(bool getTempBool){
-  float thermal_val;
-  if (isnan(powerTemperatureOverloadValue)) {
-    thermal_val =  50.0 + static_cast <float> (rand()) /( static_cast <float> (RAND_MAX/(90.0-50.0)));
-    powerTemperatureOverloadValue = thermal_val;
-  } else if (!getTempBool) {
-    powerTemperatureOverloadValue = NAN;
+void FaultInjector::publishPowerSystemFault(){
+  //power
+  ow_faults::PowerFaults power_faults_msg;
+  //system
+  std::bitset<10> systemFaultsBitmask{};
+  ow_faults::SystemFaults system_faults_msg;
+
+  systemFaultsBitmask |= m_system_faults_bitset;
+  //update if fault
+  if (m_temperature_fault || m_soc_fault) {
+    //system
+    systemFaultsBitmask |= isPowerSystemFault;
+    m_system_faults_bitset = systemFaultsBitmask;
+    //power
+    setComponentFaultsMessage(power_faults_msg, ComponentFaults::Hardware);
   }
+  //publish
+  setBitsetFaultsMessage(system_faults_msg, systemFaultsBitmask);
```

Comment:
I still think it can be safely reduced to:
```suggestion
  ow_faults::PowerFaults power_faults_msg;
  ow_faults::SystemFaults system_faults_msg;

  //update if fault
  if (m_temperature_fault || m_soc_fault) {
    //system
    m_system_faults_bitset |= isPowerSystemFault;
    //power
    setComponentFaultsMessage(power_faults_msg, ComponentFaults::Hardware);
  }
  //publish
  setBitsetFaultsMessage(system_faults_msg, m_system_faults_bitset);
```

Reply:
ah yes the version without the first declaration makes more sense